hyperspy/drawing/utils.py

```diff
@@ -1081,6 +1081,10 @@ def plot_spectra(
                            line_style=line_style,)
         if legend is not None:
             plt.legend(legend, loc=legend_loc)
+            l = ax.get_legend()
```

Comment:
Put this in an function with a comment explaining what is it doing would make the code slightly more readable?